src/main/java/com/coreos/jetcd/EtcdWatch.java

```diff
@@ -1,74 +1,81 @@
-   * might be canceled from the server-side and the onCreateFailed will be called.
+   * Watch watches on a key or prefix and returns a watcher completable future.
+   *
+   * <p>watched events can be obtained through the watcher.</p>
+   *
+   * <p>CompletableFuture watcher can return Watch specific exception.
    *
-   * @param key the key subscribe
-   * @param watchOption key option
-   * @param callback call back
-   * @return ListenableFuture watcher
+   * <pre>
+   * {@code
+   *  CompletableFuture<Watcher> watcherFuture = watchClient.watch(key, WatchOption.DEFAULT);
+   *  Watcher watcher;
+   *  try {
+   *   watcher = watcherFuture.get(100, TimeUnit.MILLISECONDS);
+   *   } catch (ExecutionException e) {
+   *    Throwable t = e.getCause();
+   *    if (t instanceof WatchCreateException) { // creating Watcher has failed.
+   *    }
+   *  } catch (TimeoutException e) { // creating watcher has timed out.
+   *  } catch (Exception e) {
+   *  }
+   * }
+   * </pre>
+   *
+   * @param watchOption see {@link com.coreos.jetcd.options.WatchOption}.
+   * @throws WatchClosedException if watch client is already closed.
    */
-  CompletableFuture<Watcher> watch(ByteSequence key, WatchOption watchOption,
-      WatchCallback callback);
+  CompletableFuture<Watcher> watch(ByteSequence key, WatchOption watchOption);
 
+  /**
+   * Close closes all watch requests.
+   */
+  void close();
+
+  /**
+   * Interface of Watcher.
+   */
   interface Watcher extends Closeable {
 
     /**
-     * get watcher id.
-     *
-     * @return id
+     * Close this watcher.
      */
-    long getWatchID();
-
-    long getLastRevision();
+    void close() throws IOException;
 
-    ByteSequence getKey();
-
-    boolean isResuming();
-
-    /**
-     * get the watch option.
-     *
-     * @return watch option
-     */
-    WatchOption getWatchOption();
 
     /**
-     * cancel the watcher.
+     * Retrieves next watch key, waiting if there are none.
      *
-     * @return cancel result
+     * @throws InterruptedException if interrupted while waiting.
+     * @throws ConnectException if watch client experiences unrecoverable connection error.
      */
-    CompletableFuture<Boolean> cancel();
-  }
-
-  interface WatchCallback {
+    WatchResponse take() throws InterruptedException, ConnectException;
 
     /**
-     * onWatch will be called when watcher receive any events.
+     * Retrieves next watch key, waiting up to the specified wait time.
      *
-     * @param events received events
-     */
-    void onWatch(EtcdHeader header, List<WatchEvent> events);
-
-    /**
-     * onResuming will be called when the watcher is on resuming.
+     * @throws InterruptedException if interrupted while waiting.
+     * @throws ConnectException if watch client experiences unrecoverable connection error.
+     * @throws TimeoutException if watcher timeout on poll.
      */
-    void onResuming();
+    WatchResponse poll(long timeout, TimeUnit unit)
```

Comment:
how about removing take? only keep poll?

also poll only takes one parameter as duration in nanoseconds?